api/mapping/management/commands/add_datasets_to_partner.py

```diff
@@ -0,0 +1,54 @@
+from django.core.management.base import BaseCommand
+from mapping.models import DataPartner, Dataset, ScanReport, Project
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        """Add `data_partner`, and `datasets` args."""
+        parser.add_argument(
+            "--data_partner",
+            required=False,
+            type=str,
+            default="None",
+            help='The data partner that owns the dataset(s). Default: "None"',
+        )
+        parser.add_argument(
+            "datasets",
+            default=[],
+            nargs="*",
+            help="""(Optional) The names of the dataset(s) to add to the data partner.
+            By default find all datasets with no data partner and add them to the specified
+            data partner or to the data partner of the dataset's first scan report.
+            """,
+        )
+
+    def handle(self, *args, **options):
+        """Logic for finding unpartnered Datasets
+        and adding them to a Data Partner.
+        """
+
+        data_partner_name = options.get("data_partner")
+        dataset_names = options.get("datasets")
+
+        print(f"Datasets will be added to Data Partner: {data_partner_name}")
+
+        if dataset_names:
+            # Get only the specified datasets
+            print("Fetching the specified datasets...")
+            orphaned_datasets = Dataset.objects.filter(name__in=dataset_names)
+        else:
+            # Get only the datasets with no data partner
+            print("No datasets given. Fetching the unpartnered datasets...")
+            orphaned_datasets = Dataset.objects.filter(data_partner=None)
+
+        # Attach datasets to the data partner
+        print(f"Attaching datasets to Data Partner: {data_partner_name}")
+        for dataset in orphaned_datasets:
+            data_partner, _ = DataPartner.objects.get_or_create(name=data_partner_name)
```

Comment:
Please move this to line 53 as we only need to create this if we ever reach a `Dataset` without an `SR`. Otherwise, we don't want to create this default DP.